src/Database/Database.php

```diff
@@ -412,6 +412,150 @@ public function deleteCollection(string $id): bool
+     * @param bool $required
+     * @param array|bool|callable|int|float|object|resource|string|null $default
+     * @param bool $signed
+     * @param bool $array
+     * @param string $format optional validation format of attribute
+     * @param string $formatOptions assoc array with custom options that can be passed for the format validation
+     * @param array $filters
+     * 
+     * To update attribute key (ID), use renameAttribute instead.
+     * 
+     * @return bool
+     */
+    public function updateAttribute(string $collection, string $id = null, string $type = null, int $size = null, bool $required = null, $default = null, bool $signed = null, bool $array = null, string $format = null, array $formatOptions = null, array $filters = null): bool
+    {
+        if($array !== null) {
+            throw new Exception('Changing array status is not supported');
+        }
+
+        // Can get to false during function
+        $success = true;
+
+        $collection = $this->getCollection($collection);
+
+        $attributes = $collection->getAttribute('attributes', []);
+
+        $attributeIndex = \array_search($id, \array_map(function($attribute) {
+            return $attribute['$id'];
+        }, $attributes));
+
+        if($attributeIndex === false) {
+            throw new Exception('Attribute not found');
+        }
+
+        $attribute = $attributes[$attributeIndex];
+
+        if($type !== null || $size !== null || $signed !== null || $array !== null) {
+            $type = $type === null ? $attribute->getAttribute('type') : $type;
+            $size = $size === null ? $attribute->getAttribute('size') : $size;
+            $signed = $signed === null ? $attribute->getAttribute('signed') : $signed;
+            $array = $array === null ? $attribute->getAttribute('array') : $array;
+
+            switch ($type) {
+                case self::VAR_STRING:
+                    if ($size > $this->adapter->getStringLimit()) {
+                        throw new Exception('Max size allowed for string is: ' . number_format($this->adapter->getStringLimit()));
+                    }
+                    break;
+    
+                case self::VAR_INTEGER:
+                    $limit = ($signed) ? $this->adapter->getIntLimit() / 2 : $this->adapter->getIntLimit();
+                    if ($size > $limit) {
+                        throw new Exception('Max size allowed for int is: ' . number_format($limit));
+                    }
+                    break;
+                case self::VAR_FLOAT:
+                case self::VAR_BOOLEAN:
+                    break;
+                default:
+                    throw new Exception('Unknown attribute type: ' . $type);
+                    break;
+            }
+
+            $attribute->setAttribute('type', $type);
+            $attribute->setAttribute('size', $size);
+            $attribute->setAttribute('signed', $signed);
+            $attribute->setAttribute('array', $array);
```

Comment:
```suggestion
            $attribute
                ->setAttribute('type', $type)
                ->setAttribute('size', $size)
                ->setAttribute('signed', $signed)
                ->setAttribute('array', $array);
```

this can be chained.

Reply:
Applied